src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -322,9 +346,8 @@ internal async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(bool
             else
             {
                 return targets
-                    .SelectMany(target => target.Libraries)
-                    .Where(library => library.Type == LibraryType.Package)
-                    .SelectMany(library => GetPackageReferenceUtility.UpdateTransitiveDependencies(library, targetFramework, targets, installedPackages, transitivePackages))
+                    .Where(target => target.TargetFramework.Equals(targetFramework))
+                    .SelectMany(target => GetPackageReferenceUtility.UpdateTransitiveDependencies2(target.Libraries, installedPackages, transitivePackages))
```

Comment:
We could also use `LockFile.Libraries` instead of `LockFile.Targets` to remove the need of comparing to a target framework since we don't handle tfm's in the PM UI but I will explore that idea in another PR.

Reply:
Libraries section won't have the dependencies.

Reply:
Truee, then I'm not sure if it's worth loading Libraries section